src/js/utils/date-utils.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Create a formatted, locale-aware string representation of a datetime.
+ *
+ * This module contains general purpose functions for localizing and displaying
+ * date-times by timezone, locale and preferred language."
+ *
+ * Most of the heavy lifting is done by the {@link http://momentjs.com/ |'moments' js library}
+ * with the moment-with-locales and moment-timezones add-ons.
+ *
+ *
+ *
+ * Example:
+ *
+ *~~~ javascript
+ * var context = {
+ *    datetime: '2016-10-14 08:00:00+00:00',
+ *    timezone: 'Pacific/Kiritimati',
+ *    language: 'ar',
+ *    format: 'DEFAULT_LONG_DATE_FORMAT'
```

Comment:
Per our offline conversation, consider changing the interface so callers can pass in an enum constant instead of a string - this allows issues in the caller to be caught by the interpreter - and not silently fail.
This change can be made as part of this PR or a separate PR.